### DISCUSSION TITLE
Smallrye fault tolerance mocking

### DISCUSSION THREAD

**Discussion by nshojaedin** [quarkusio/quarkus#32808] [Q&A]
I am trying to unit test a method which has Fallback, Timeout, and Circuit Breaker fault tolerance resiliencies. I am wondering if mocking faulty behavior to simulate these annotations is possible at all?

[a comment by an automated account is not shown]

**Comment by Ladicek** [comment]
I assume you want to _trigger_ the fault and hence test that fault tolerance works as expected. I'm afraid that is not possible at the moment, and I'm not sure how could something like this be made. If you have any ideas how an API for "mocking the faulty behavior" would look like, I'm all ears -- maybe we can figure something out.
